### PR TITLE
Add display next event dojo listing

### PR DESCRIPTION
**Pull request by Wardormeur**
## Description
Add a display of the next event date on the dojo search page to engage users to click the dojo link when it makes sense (an event to come)
Only works for non-private dojos and non-private events.

## Motivation and Context

Fixes https://trello.com/c/fhcHn1h1

## How Has This Been Tested?

Mock data on chrome

## Screenshots (if appropriate):
see ticket

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I gave a thought about security (XSS, perms, injection of params, scoping, ...)
- [X] Linting passes

## Related PRs

https://github.com/CoderDojo/cp-translations/pull/52

## Reviewer checklist:

 - [ ] Security
   - [ ] Permissions
   - [ ] Data scope and leaks
   - [ ] Injection
   - [ ] XSS
 - [ ] Performance
   - [ ] tested with production data
 - [ ] Code
   - [ ] Single responsibility
   - [ ] Naming
 - [ ] Design
   - [ ] Responsive
   - [ ] Scale well with production data
   - [ ] Looks great !

### CODE REVIEW

**Comment by Wardormeur** [comment]
Missing : filter on status = published

File: src/events/cd-event-stamp.vue [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    },
+    async created() {
+      const events = await EventService.v3.get(
+        this.dojo.id,
+        {
+          params:
+          {
+            limit: 1,
+            public: 1,
+            dateAfter: Date.now(),
```
Comment (Wardormeur) [inline src/events/cd-event-stamp.vue]: Date.now is not a valid unix timestamp. Use moment().format('x') instead

File: src/events/cd-event-stamp.vue [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    data() {
+      return {
+        event: null,
+      };
+    },
+    computed: {
+      date() {
+        const toBeFormatted = (this.event.startTime) ||
+          this.event.dates[0].startTime;
+        return moment(toBeFormatted).format('dddd, MMMM Do');
```
Comment (DanielBrierton) [inline src/events/cd-event-stamp.vue]: `moment.utc(toBeFormatted)` because timezones

File: src/events/cd-event-stamp.vue [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      {{ $t('Next event on {date}', { date }) }}
+  </router-link>
+</template>
+<script>
+  import moment from 'moment';
+  import DojosUtil from '@/dojos/util';
+  import EventService from './service';
+
+  export default {
+    name: 'event-stamp',
```
Comment (DanielBrierton) [inline src/events/cd-event-stamp.vue]: ew.. Maybe I'll have an epiphany on the name over the weekend 😛

File: src/events/cd-event-stamp.vue [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      const events = await EventService.v3.get(
+        this.dojo.id,
+        {
+          params:
+          {
+            pageSize: 1,
+            query: {
+              status: 'published',
+              public: 1,
+              afterDate: moment().format('X'),
```
Comment (DanielBrierton) [inline src/events/cd-event-stamp.vue]: `moment().unix()`

File: src/events/service.js [hunk 5]
```diff
@@ -30,6 +30,11 @@ const EventsService = {
     });
     return Vue.http.post(`${Vue.config.apiServer}/api/2.0/events/bulk-apply-applications`, { applications });
   },
+  v3: {
```
Comment (DanielBrierton) [inline src/events/service.js]: Woooooooo v3!

**Pull request merged**